ext/MitsubaLoader/Shape.h

```diff
@@ -12,7 +12,7 @@ class CShape : public IElement
 	CShape()
 		: mesh(nullptr) {};
 
-	inline asset::ICPUMesh* getMesh() { return mesh; }
+	inline core::smart_refctd_ptr<asset::ICPUMesh> getMesh() { return mesh; }
```

Comment:
the convention here is to return the raw pointer, we only want to return a `smart_refctd_ptr` if we are sure that the callee wants to hold onto the reference for a longer time.